Test/FMI3/fmi3_basic_capi_test.cpp

```diff
@@ -0,0 +1,524 @@
+        int expected_mode = 1;
+        int actual_mode = -1; // value will be overriden
+        // it is required to setup the C-API struct before invoking set/get debug mode
+        jm_status_enu_t status = fmi3_import_create_dllfmu(fmu, fmi3_fmu_kind_se, &inst_env, fmi3_dummy_log_callback);
+        REQUIRE(status == 0);
+
+        fmi3_import_set_debug_mode(fmu, expected_mode);
+        actual_mode = fmi3_capi_get_debug_mode(fmu->capi);
+        REQUIRE(actual_mode == expected_mode);
+        fmi3_import_set_debug_mode(fmu, 0); // set back to zero to prevent segfaults
+    }
+
+    SECTION( "Test instantiate and set values on float64 array" ) {
+        size_t n_value_references = 1;
+        const fmi3_value_reference_t value_references[n_value_references] = {1234567};
+        size_t n_values = 4;
+        const fmi3_float64_t values[n_values] = {1.2, 2.5, 3.2, 4.5};
+        jm_status_enu_t status = fmi3_import_create_dllfmu(fmu, fmi3_fmu_kind_se, &inst_env, fmi3_dummy_log_callback);
+        REQUIRE( status == 0 );
+        fmi3_string_t instanceName = "Test SE model instance";
+        fmi3_string_t fmuInstantiationToken;
+        fmi3_string_t resourcePath = "";
+        fmi3_boolean_t visible = fmi3_false;
+        fmi3_boolean_t loggingOn = fmi3_false;
+        dummy_fmi3_instance_environment_t instance_env = {};
+        jm_status_enu_t instantiate_status = fmi3_import_instantiate_scheduled_execution(
+                fmu,
+                instanceName,
+                resourcePath,
+                visible,
+                loggingOn,
+                ((fmi3_instance_environment_t)&instance_env),
+                (fmi3_log_message_callback_ft)dummy_log_message_callback,
+                (fmi3_clock_update_callback_ft)dummy_clock_update_callback,
+                (fmi3_lock_preemption_callback_ft)dummy_lock_preemption_callback,
+                (fmi3_unlock_preemption_callback_ft)dummy_unlock_preemption_callback
+        );
+        REQUIRE( instantiate_status == jm_status_success );
+        fmi3_status_t status_from_float_set = fmi3_import_set_float64(
+            fmu,
+            value_references,
+            n_value_references,
+            values,
+            n_values
+        );
+        REQUIRE( status_from_float_set == fmi3_status_ok );
+
+        fmi3_float64_t values_from_get[n_values] = {0.0, 0.0, 0.0, 0.0};
+        fmi3_status_t status_from_float_get = fmi3_import_get_float64(
+            fmu,
+            value_references,
+            n_value_references,
+            values_from_get,
+            n_values
+        );
+        REQUIRE( status_from_float_get == fmi3_status_ok );
+        // We use indivdual calls to REQUIRE to make a potential error point to the exact line
+        REQUIRE( values[0] == values_from_get[0] );
+        REQUIRE( values[1] == values_from_get[1] );
+        REQUIRE( values[2] == values_from_get[2] );
+        REQUIRE( values[3] == values_from_get[3] );
```

Comment:
This code is duplicated like 3 times. You can just move it to a static function.

Reply:
Fixed!